### PR TITLE
Retrieve student respository

### PR DESCRIPTION
**Pull request by brominger**
I wrote the StudentInfoRepositoryImpl class. Which searches through the list of student records using a student ID and returns a StudentInfo object if found and null if not found.
I created a Junit test for the getInfo method and added javadocs.

Ehsan

I know you wants us to use org.json instead of org.json.simple but I could not find the JSONParser method in org.json. I sure there is another method that does the same thing. If we have time at the end I will rewrite this

### CODE REVIEW

**Comment by samaniehsan** [comment]
@brominger we need to do a rebase to get your changes in (There is a conflict since i changed it before you).
I can do it. but if you want, I can delay it until we are in class to show you how to resolve conflicts with git.

let me know, which one you prefer.

File: src/main/java/auto_scheduler/StudentInfoRepositoryImpl.java [hunk 1]
```diff
@@ -1,7 +1,62 @@
...
+     * Searches json files in records by studentID
+     * If found return StudentInfo object
+     * If not found returns null
+     * @param studentId
+     * @return StudentInfo
+     */
+    public StudentInfo getInfo(String studentId) {
+
+        JSONParser parser = new JSONParser();
+        File folder = new File("src/main/resources/data/records/");
```
Comment (samaniehsan) [inline src/main/java/auto_scheduler/StudentInfoRepositoryImpl.java]: we need to use the `ResourcePathProvider` interface to get the path of records folder. 
see sample implementation from [StudentRecordRepositoryImpl](https://github.com/samaniehsan/auto-scheduler/blob/master/src/main/java/auto_scheduler/StudentRecordRepositoryImpl.java)


```java
public StudentRecordRepositoryImpl(ResourcePathProvider pathProvider) {
		this.pathProvider = pathProvider;
	}
	
	public Collection<StudentRecord> get(String studentId) throws FileNotFoundException,  IOException {
		return getGrades(getFilePath(studentId), studentId);
	}


	private String getFilePath(String studentId) {
		String recordsFolderRelativePath = this.pathProvider.getRecords();
		String gradeCSVPath = recordsFolderRelativePath + "/" + studentId + "_grade.csv";
		return gradeCSVPath;

}
```

Comment (brominger) [reply src/main/java/auto_scheduler/StudentInfoRepositoryImpl.java]: I will be in class tonight. So, you can show me how to rebase it. I also will implement the ResourcePathProvider interface it when I get a chance.

**Comment by samaniehsan** [comment]
@brominger 

my commits explained

- Commit To Compile Unit Tests
- Commit to Fix Unit tests at run time
- Commit to Fix Efficiency (not running the entire folder). directly go to json file.

**Pull request merged**